### ISSUE TITLE
feat: Readonly attribute for ion-checkbox, ion-toggle, ion-radio

### ISSUE THREAD

**Issue by fudom** [ionic-team/ionic-framework#29603]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/docs/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already include this feature request, without success.

### Describe the Feature Request

I updated Ionic from 6 (to 7) to 8 and have problems with the breaking changes of the input components. Feedback: I'm not sure if the new label attribute solution is better than the ion-label before. It restricts a lot of features like text-wrap and more complex labels like sub-label etc. If I use the legacy ion-label, it blocks the click event and you cannot toggle the checkbox anymore... I'm sure you had good reasons for that, but it feels like one step forward and two steps backward. Anyway...

The reason for this ticket is the following problem:

The new behavior of ion-item with an input element like e.g. ion-toggle is now different:

- Ionic 6: The click event of ion-item didn't affected the input element inside.
- Ionic 8: The click event of ion-item is directed to the input. Item click always toggles checked.

What's the problem? Well, it's ok. But you can't control the ion-toggle (or checkbox whatever) indirectly anymore. Let me show an use case.

### Describe the Use Case

```html
<ion-item button (click)="toggleValue()">
  <ion-toggle class="no-pointer-events" [checked]="value" justify="space-between">
    Toggle Me
  </ion-toggle>
</ion-item>
```

The code above is a migration from Ionic 6 to 8. I use the new label style. Well, looks good. But the problem is, that (click) on item changes the toggle value. In this case, I don't want to change the value directly. Therefore I added `pointer-event: none` to it. Works great in Ionic 6. Ionic 8 does not care about. Because ion-items handles it now.

Background: The function `toggleValue` on click calls and changes the `value` if possible. But now the binding `checked` is ignored and the toggle is always toggles. Or toggles twice if the value is toggles indirectly when clicked. 

### Describe Preferred Solution

I want disconnect the input element like `ion-toggle` or `ion-checkbox` from `ion-item` to use the click event without affecting the input element. Like `stopPropagation` or `preventDefault` which I already tried. Maybe add an attribute `mute` or `passive` to the input? Btw. `readonly` does also not work and `disabled` would change the appearence and would the click event would not work.

In other words, in this case: The ion-toggle etc should only be used as indicator. A dummy only for showing the current state. It should not be controlled directly by click. Only indirectly by item click event. We could implement `readonly` which is the same as disabled but appears regular (not grayed out).

My current workaround: Use `disabled` and a class name to set the opacity back to 1. Fortunately, the click event of ion-item still works. It seems to work as expected. So readonly attribute may fix this issue.

**Comment by liamdebeasi** [comment]
This might be the same request as https://github.com/ionic-team/ionic-framework/issues/20521 (just with checkbox instead of toggle)

> It restricts a lot of features like text-wrap and more complex labels like sub-label etc.

You can absolutely still do this with the new syntax. Check out https://ionicframework.com/docs/api/item#text (this example uses a toggle with wrapping/subtext, but it works the same way with checkboxes).

**Comment by liamdebeasi** [comment]
Also I wouldn’t recommend the usage you are proposing as it creates a nested interactive. A nested interactive is a bad practice that makes your app less accessible because screen readers don’t always know which interactive to focus. In this case there is a checkbox inside of a button. 

If you want to indirectly control the checkbox using another button in the item I’d try putting the checkbox in the start slot and putting the toggle button in the default slot.

**Comment by liamdebeasi** [comment]
>  If I do so, it toggles twice which appears as not change

Inside of `ion-toggle` is a `label`. When a click event is dispatched on that element (by the user clicking) it will dispatch another click event on any descendant inputs. Both of these click events bubble up across the shadow dom boundary. You can filter out the duplicate click event by ignoring ones with `pointerId: -1` 

> The toggle is not used as input directly. It's more like an icon, an indicator. It cannot be changed directly. Therefore I used the layout like from Ionic 6 with item slots. But I could also use the new label style. It does not really matter for this issue/case. It's about readonly input.

I'm not entirely sure why you are wanting to do this as it goes against both iOS and Android design principles. Even if you removed the pointer events from the toggle screen readers would still read it as a toggle instead of an icon which could confused users.

**Comment by liamdebeasi** [comment]
Also your current solution is confusing because screen readers will inform users that the toggle is not interactive because it is disabled. However, clicking on the item will still update the toggle.

**Comment by liamdebeasi** [comment]
Screen readers exist on mobile devices (VoiceOver on iOS and TalkBack on Android) and can be used with Capacitor/Cordova apps.

Doing something like this will give you an experience that is accessible and familiar to most users: https://codepen.io/liamdebeasi/pen/pomWpZr

**Comment by thetaPC** [comment]
Closing this issue due to inactivity. The suggested solutions should resolve the problem. If you’re still experiencing issues, feel free to open a new one.
